source/particles/particle_handler.cc

```diff
@@ -1320,26 +1358,56 @@ namespace Particles
         &vertex_to_cell_centers =
           triangulation_cache->get_vertex_to_cell_centers_directions();
 
-      std::vector<unsigned int> search_order;
+      // Now in the next step we find where the particles that have left their
+      // cell have gone to. They may be lost entirely or located in a new cell.
+      // We can try to find the new cells of multiple particles in parallel,
+      // however we cannot reinsert the particles until we have finished looking
+      // at all of them or we risk invalidating the data the worker is looking
+      // at.
```

Comment:
A suggestion for wording:

```suggestion
      // In stage two we search for the new cells of particles that have left their old cell.
      // The particles may be located in a new cell or be lost entirely if they left the triangulation.
      // We try to find the new cells of multiple particles in parallel worker threads,
      // however we cannot reinsert the particles until we have finished looking
      // at all of them or we risk invalidating the data the worker is looking
      // at.
```

However, I do not understand the problem with the queuing. The algorithm should for each particle find the correct new cell. Then it inserts the algorithm into the new cell. I do not see where it access information about particles in the new cell, except during the insertion (which could be serial in the copier). Please extend the comment to be more specific on which data would be invalidated if you reinsert the particle already.